database/schema.sql

```diff
@@ -500,3 +500,44 @@ update dbo.sccv_personal_relationship_type set inverse_type_id = (select id from
 
 ALTER TABLE dbo.sccv_personal_relationship
   ADD COLUMN is_main_carer varchar(1);
+
+--- deleted person records handling
+ALTER TABLE dbo.dm_persons
+  ADD COLUMN marked_for_deletion boolean;  
+  
+ALTER TABLE dbo.dm_addresses
+  ADD COLUMN marked_for_deletion boolean; 
+  
+ALTER TABLE dbo.dm_telephone_numbers
+  ADD COLUMN marked_for_deletion boolean; 
+  
+ALTER TABLE dbo.sccv_person_other_name
+  ADD COLUMN marked_for_deletion boolean; 
+  
+ALTER TABLE dbo.sccv_allocations_combined
+  ADD COLUMN marked_for_deletion boolean; 
+  
+ALTER TABLE dbo.sccv_warning_note
+  ADD COLUMN marked_for_deletion boolean; 
+  
+ALTER TABLE dbo.sccv_personal_relationship
+  ADD COLUMN marked_for_deletion boolean;   
```

Comment:
Just thinking do we also need to add this to the `sccv_personal_relationship_detail`? 🤔

Reply:
These flags are only added to entities directly linked to Person entity. For example I haven't included warning note reviews. Those will have to be included in the hard delete script, but I thought they are not necessary for the initial soft delete. Hard delete script will scan through the entity relationships and delete them all.